src/transformers/modeling_tf_utils.py

```diff
@@ -684,28 +713,76 @@ def _get_resized_embeddings(self, old_embeddings, new_num_tokens=None) -> tf.Var
+        bias_layer = self.get_output_layer_with_bias()
+
         if new_num_tokens is None:
             return word_embeddings
+
         old_num_tokens, old_embedding_dim = word_embeddings.shape
+
         if old_num_tokens == new_num_tokens:
             return word_embeddings
 
         # initialize new embeddings
         # todo: initializer range is not always passed in config.
         init_range = getattr(self.config, "initializer_range", 0.02)
+        name = (
+            self.name
+            + "/"
+            + self.base_model_prefix
+            + "/"
+            + old_embeddings.name
+            + "/"
+            + word_embeddings.name.split(":")[0]
+        )
         new_embeddings = self.add_weight(
-            "weight",
+            name=name,
             shape=[new_num_tokens, old_embedding_dim],
             initializer=get_initializer(init_range),
             dtype=tf.float32,
         )
-        init_weights = new_embeddings.numpy()
+        init_weights = tf.make_ndarray(tf.make_tensor_proto(new_embeddings.value()))
 
         # Copy token embeddings from the previous weights
         num_tokens_to_copy = min(old_num_tokens, new_num_tokens)
-        init_weights[:num_tokens_to_copy] = word_embeddings[:num_tokens_to_copy, :]
+        init_weights[:num_tokens_to_copy] = word_embeddings.value()[:num_tokens_to_copy, :]
         new_embeddings.assign(init_weights)
 
+        if bias_layer is not None:
+            if not hasattr(bias_layer, "bias"):
+                bias_layer.build([])
+
+            # Second check in order to be sure the attribute has been properly created
+            if not hasattr(bias_layer, "bias"):
+                raise ValueError("bias is not defined.")
+
+            # initialize bias
+            init_bias = np.zeros((new_num_tokens,))
+            init_bias[:num_tokens_to_copy] = bias_layer.bias.value()[
+                :num_tokens_to_copy
+            ]  # tf.make_ndarray(tf.make_tensor_proto(bias_layer.bias.value()))[:num_tokens_to_copy]
+
+            bias_layer.bias = self.add_weight(
+                shape=(new_num_tokens,),
+                initializer="zeros",
+                trainable=True,
+                name=self.get_prefix_bias_name() + "/bias",
+            )
+
+            bias_layer.bias.assign(init_bias)
+
+            if self.get_input_embeddings() != self.get_output_embeddings():
+                init_weights = np.zeros((new_num_tokens, old_embedding_dim))
+                init_weights[:num_tokens_to_copy] = bias_layer.decoder.value()[:num_tokens_to_copy, :]
+
+                bias_layer.decoder = self.add_weight(
+                    shape=(new_num_tokens, old_embedding_dim),
+                    initializer="zeros",
+                    trainable=True,
+                    name=self.get_prefix_bias_name() + "/decoder/weight",
+                )
+                bias_layer.decoder.assign(init_weights)
```

Comment:
Should we use the test with `get_output_layer_with_bias`? It seems weird to test the difference with two layers that are then never used. Also I was thinking we could have differences without a bias, so this test could be independent of the `bias_layer is not None` test?

Reply:
You mean to use the returned value of `get_output_embeddings` instead of the one of `get_output_layer_with_bias` ?

>Also I was thinking we could have differences without a bias, so this test could be independent of the bias_layer is not None test?

Yes! Making an update.

Reply:
Yes that would make more sense to me.

Reply:
Does-it looks better for you now?